pages/create-issue.tsx

```diff
@@ -45,14 +61,50 @@ export default function PageCreateIssue() {
 
   }
 
+  const titleAndIssueExist = () => {
+    return !(issueTitle.length && issueDescription.length)
+  }
+
+  const verifyAmountBiggerThanBalance = () => {
+    return !(issueAmount.floatValue > Number(balance))
+  }
+
+  const checksToEnableCreateIssue = () => {
+    if (!allowedTransaction){
```

Comment:
you don't need to `else if` if you have returned from the statement,

```ts
checksToEnableCreateIssue = () => {
  if (!allowedTransaction)
    return true
    
  if (titleAndIssueExist())
    return true

  // ...
}
```

But then again, the naming of this function is all over the place, _and_ this is used on a "disabled" prop meaning the person reading this code will have to know, beforehand, that this function does not do what it says it does.
And you can also simplify this function to its purest forms:

We know that all of these checks need to be true, otherwise the button will be disabled. meaning that if any of these values is false, we return true:

```ts
function isButtonDisabled(): boolean {
  return [
      allowedTransaction,
      titleAndIssueExist(),
      verifyAmountBiggerThanBalance(),
      issueAmount.floatValue > 0 || !!issueAmount.formattedValue
    ].some(v === false);
}
```

**NOTICE** for some reason, all the functions are in the positive, but their return value is negative;
Take this function for instance: "title and issue exist" meaning that If I have a title and a description this function should return true but it doesn't:

|combination|result|
|---|---|
|!(1 && 1)|false|
|!(1 && 0)|true|
|!(0 && 1)|true|
|!(0 && 0)|true|

```ts
  const titleAndIssueExist = () => {
    return !(issueTitle.length && issueDescription.length)
  }
 ```
 
to make it easy on everyone I'd simply rewrite all the functions to be _affirmitive_ and return _true_:

```
const issueContentIsValid = () =>  !!issueTitle && !!issueDescription;
```

And then I would do the same for every "is X valid" function

Reply:
Good. Adjustments done!